alibi_detect/cd/classifier.py

```diff
@@ -1,215 +1,72 @@
-
-        Parameters
-        ----------
-        threshold
-            Threshold for the drift metric (default is accuracy). Values above the threshold are
-            classified as drift.
-        model
-            Classification model used for drift detection.
-        X_ref
-            Data used as reference distribution. Can be a list for text data which is then turned into an array
-            after the preprocessing step.
-        preprocess_X_ref
-            Whether to already preprocess and store the reference data.
-        update_X_ref
-            Reference data can optionally be updated to the last n instances seen by the detector
-            or via reservoir sampling with size n. For the former, the parameter equals {'last': n} while
-            for reservoir sampling {'reservoir_sampling': n} is passed.
-        preprocess_fn
-            Function to preprocess the data before computing the data drift metrics.
-        preprocess_kwargs
-            Kwargs for `preprocess_fn`.
-        metric_fn
-            Function computing the drift metric. Takes `y_true` and `y_pred` as input and
-            returns a float: metric_fn(y_true, y_pred). Defaults to accuracy.
-        metric_name
-            Optional name for the metric_fn used in the return dict. Defaults to 'metric_fn.__name__'.
-        train_size
-            Optional fraction (float between 0 and 1) of the dataset used to train the classifier.
-            The drift is detected on `1 - train_size`. Cannot be used in combination with `n_folds`.
-        n_folds
-            Optional number of stratified folds used for training. The metric is then calculated
-            on all the out-of-fold predictions. This allows to leverage all the reference and test data
-            for drift detection at the expense of longer computation. If both `train_size` and `n_folds`
-            are specified, `n_folds` is prioritized.
-        seed
-            Optional random seed for fold selection.
-        optimizer
-            Optimizer used during training of the classifier.
-        compile_kwargs
-            Optional additional kwargs when compiling the classifier.
-        batch_size
-            Batch size used during training of the classifier.
-        epochs
-            Number of training epochs for the classifier for each (optional) fold.
-        verbose
-            Verbosity level during the training of the classifier.
-            0 is silent, 1 a progress bar and 2 prints the statistics after each epoch.
-        fit_kwargs
-            Optional additional kwargs when fitting the classifier.
-        data_type
-            Optionally specify the data type (tabular, image or time-series). Added to metadata.
-        """
+if has_pytorch:
+    from alibi_detect.cd.pytorch.classifier import ClassifierDriftTorch
+
+if has_tensorflow:
+    from alibi_detect.cd.tensorflow.classifier import ClassifierDriftTF
+
+
+class ClassifierDrift:
```

Comment:
We should probably still make this inherit from `BaseDetector` to adhere to a well-defined interface, in particular it should have a `meta` property (which can just dispatch to the `_detector.meta`. I note that the `score` method in `BaseDetector` may not apply here (as the backend takes care of it), does it make sense to remove `score` from `BaseDetector` entirely then?

Reply:
Agreed to address in follow up PR.